File: noop-java/src/main/java/com/exyui/android/debugbottle/components/Injector/QuickEntry.java
package com.exyui.android.debugbottle.components.Injector;

import android.app.Activity;
import android.content.Context;
import android.util.Log;

import java.util.LinkedHashMap;

/**
 * Created by yuriel on 10/17/16.
 */

@SuppressWarnings("unused")
public class QuickEntry {
    private final String TAG = "QuickEntry";
    public void put(Object a, Object b, Object c) {
        Log.d(TAG, "put");
    }
    public void put(Object a, Object b) {
        Log.d(TAG, "put");
    }
    public OnActivityDisplayedListener get() {
        return null;
    }
    public void run(Object a) {
        Log.d(TAG, "run");
    }
    public LinkedHashMap<String, OnActivityDisplayedListener> getList() {
        return null;
    }
    interface OnActivityDisplayedListener {
        boolean shouldShowEntry(Activity activity);
        void run(Context context);
    }
}

File: noop-kotlin/src/main/kotlin/com/exyui/android/debugbottle/components/injector/QuickEntry.kt
@file:Suppress("unused", "UNUSED_PARAMETER")

package com.exyui.android.debugbottle.components.injector

import android.app.Activity
import android.content.Context
import android.util.Log
import java.util.*

/**
 * Created by yuriel on 10/17/16.
 */
object QuickEntry {
    private val TAG = "QuickEntry"

    fun put(a: Any?, b: Any?, c: Any? = null) {
        Log.d(TAG, "put")
    }

    fun get(): OnActivityDisplayedListener? {
        return null
    }

    fun run(a: Any?) {
        Log.d(TAG, "run")
    }

    fun getList(): LinkedHashMap<String, OnActivityDisplayedListener>? {
        return null
    }

    internal fun isEmpty() = true

    interface OnActivityDisplayedListener {
        fun shouldShowEntry(activity: Activity?): Boolean
        fun run(context: Context?)
        fun description(): String?
    }
}